### PR TITLE
Validate section navigation

### PR DESCRIPTION
**Pull request by tgandrews**
### What is the context of this PR?
Add first section validation which validates that a section title is provided when navigation is enabled.

### How to review 
1. Ensure that when section navigation is enabled against the questionnaire then an error is shown for sections without titles
1. Ensure that the error is removed/added when the title is changed

### CODE REVIEW

**Comment by samiwel** [comment]
[Validate Section Navigation (6)](https://trello.com/c/YqevMFlK/1174-validate-section-navigation-6)

File: eq-author-api/src/validation/schemas/entities.json [hunk 1]
```diff
@@ -6,7 +6,8 @@
       "properties": {
         "id": { "type": "string" },
         "title": {
-          "type": "string"
+          "type": "string",
+          "requiredWhenQuestionnaireSetting": "navigation"
```
Comment (samiwel) [inline eq-author-api/src/validation/schemas/entities.json]: Nice use of the custom keyword 👍

File: eq-author/src/App/section/Design/SectionEditor/SectionEditor.test.js [hunk 2]
```diff
@@ -136,6 +146,28 @@ describe("SectionEditor", () => {
...
+    expect(
+      wrapper
+        .find("[testSelector='txt-section-title']")
+        .prop("errorValidationMsg")
+    ).toBe("Validation error");
+
+    expect(getValidationError).toHaveBeenCalledWith({
+      field: "title",
+      message:
+        "Enter a section title. If section navigation is not required, disable in 'settings'.",
```
Comment (samiwel) [inline eq-author/src/App/section/Design/SectionEditor/SectionEditor.test.js]: Seems weird to be calling `getValidationError` passing in the message, only to return the message, but I appreciate why it's being done.

File: eq-author-api/src/validation/index.test.js [hunk 3]
```diff
@@ -71,4 +71,42 @@ describe("schema validation", () => {
...
+      questionnaire.navigation = false;
+      const section = questionnaire.sections[0];
+      section.title = "";
+
+      const validationErrors = validation(questionnaire);
+      const sectionErrors = validationErrors.sections[section.id];
+      expect(sectionErrors).toBeUndefined();
+    });
+
+    it("should NOT return an when navigation is enabled and there is a title", () => {
```
Comment (heofs) [inline eq-author-api/src/validation/index.test.js]: return an error?

File: eq-author/src/App/section/Design/SectionEditor/index.js [hunk 4]
```diff
@@ -139,6 +142,11 @@ class SectionEditor extends React.Component {
               size="large"
               testSelector="txt-section-title"
               autoFocus={autoFocusTitle}
+              errorValidationMsg={this.props.getValidationError({
+                field: "title",
+                message:
+                  "Enter a section title. If section navigation is not required, disable in 'settings'.",
```
Comment (heofs) [inline eq-author/src/App/section/Design/SectionEditor/index.js]: Should it be `disable it in 'settings'.`?

Comment (tgandrews) [reply eq-author/src/App/section/Design/SectionEditor/index.js]: The text came from the card

**Pull request merged**